### PR TITLE
Add three dot menu to the post [ Report post ]

### PR DESCRIPTION
**Pull request by anikakash**
**Bug Fix:**
1. Authorized users can now update or delete only their own posts.

**New Feature:**
1. Implemented a reporting system for posts by other users.

### CODE REVIEW

**Review by anikakash — COMMENTED** [review]
Review done from my side

File: server/feed/serializers.py [hunk 1]
```diff
@@ -36,6 +36,18 @@ def create(self, validated_data):
...
+class ReportSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Report
+        fields = ['id', 'type']  # 'post' and 'user' are excluded as they are handled in the view
+
+    def validate(self, data):
+        request = self.context['request']
+        post = self.context['post']
+        if post.author == request.user:
+            raise serializers.ValidationError("You cannot report your own post.")
```
Comment (srijonashraf) [inline server/feed/serializers.py:48]: validation could be handled from View

**Review by srijonashraf — CHANGES_REQUESTED** [review]

File: server/feed/serializers.py (hunk 1 above)
Comment (anikakash) [reply server/feed/serializers.py]: done

**Review by anikakash — COMMENTED** [review]
made request review changes.

**Review by srijonashraf — APPROVED** [review]
Approved

**Pull request merged**